### DISCUSSION TITLE
"kops" cluster, Failed state after 24/48h

### DISCUSSION THREAD

**Discussion by voriol** [argoproj/argo-cd#10770] [Q&A]
When i add a "kops" cluster to the "argocd" clusters lists, at beginning all is OK, i can list the cluster, deploy applications, etc.

But, at 24/48h, a message appears in the cluster lists: "the server has asked for the client to provide credentials". And the status of the cluster is: Failed.

I have 3 clusters more, non kops clusters, and i don't have this problem.

Any ideas? Thank you so much.

**Comment by jgwest** [comment]
When adding a cluster via Argo CD CLI's `cluster add` command, Argo CD creates a service account (and corresponding Secret containing service account token) which it uses to deploy to the cluster. It sounds like the ServiceAccount token is being invalidated on some fixed daily schedule. This would be a policy of the Kubernetes cluster, rather than anything specific to Argo CD. (I'm not specifically familiar with kops.)

More on how cluster secrets are stored by Argo CD: https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#clusters
